### DISCUSSION TITLE
User notifications / Уведомления пользователей

### DISCUSSION THREAD

**Discussion by paul-k-pastvu** [PastVu/pastvu#338] [Ideas / Идеи и предложения]
Сейчас на сайте реализовано уведомление пользователей только о комментариях:

- Через рассылку сообщений о новых комментариях к тем фото и новостям, на которые они подписаны
- через сортировку и выделение на странице подписок
- через отображение числа непрочитанных комментариев в галереях

Однако есть и другие события, о которых нужно уведомлять пользователей:

- Изменение их полномочий на сайте (получение/снятие флага модератора), а также изменение администратором настроек в их профиле (различные ограничения)
- отклонение/деактивация/отправка на доработку загруженных пользователем изображений
- упоминание никнейма пользователя в комментариях к изображениям и новостям

Опционально (по выбору пользователя) можно было бы уведомлять его о публикации фото, отправленных им на модерацию.

Для модераторов было бы полезным узнавать о новых комментариях к фотографиям в выбранных ими регионах, а также о действиях с фото в своих регионах, выполненных _другими_ модераторами.

----
На эту тему в разделе задач [есть отдельная метка "Notifications"](https://github.com/PastVu/pastvu/issues?q=is%3Aopen+is%3Aissue+label%3ANotifications)

**Comment by paul-k-pastvu** [comment]
Основные вопросы - 

- как именно принципиально будет реализована система уведомлений?
- что нужно поменять в уже сформулированных задачах?
- как сформулировать недостающие задачи для "покрытия" темы?

**Comment by paul-k-pastvu** [comment]
Отдельный вопрос - где именно пользователь должен видеть уведомления?
На мой взгляд, 

- должна быть отдельная лента уведомлений, где будет видно, что за действие совершено, когда и кем
- если у пользователя есть непросмотренные уведомления, то под шапкой сайта он должен видеть появляющуюся статусную строку с текстом примерно таким: "У вас есть фотографии, отправленные на доработку. Просмотреть..." (текст зависит от типа события)
- если у пользователя есть непросмотренные отклоненные/деактивированные фотографии, то, зайдя в свою галерею, он там должен видеть сообщение об этом со ссылкой на включение фильтра "Отклоненные"/"Неактивные"
- если у пользователя есть фотографии, отправленные на доработку (неважно, просмотренные или нет), то, зайдя в свою галерею, он там должен видеть сообщение об этом со ссылкой на включение фильтра "На доработке"

**Comment by ShockwaveNN** [comment]
> Для модераторов было бы полезным узнавать о новых комментариях к фотографиям в выбранных ими регионах, а также о действиях с фото в своих регионах, выполненных другими модераторами.

Еще как модератор региона скажу, что было бы неплохо иметь возможность получать на почту уведомление о "новых фото в очереди подтверждения своего региона"

Думаю если достаточно ежедневной рассылки "за последние сутки в очереди 2 висят фото на подтверждение"

**Reply by kabalin** [reply]
@paul-k-pastvu добавил эту задачу https://github.com/PastVu/pastvu/issues/453
